### ISSUE TITLE
Pending refunds in Stripe listed as failed in Commerce

### ISSUE THREAD

**Issue by nickcobley** [craftcms/commerce-stripe#134]
### Description

We have a customer that for various reasons is having to do a large percentage of refunds on an ongoing basis. These refunds are often over 4 days from the original transition.

The issue here is after a few days, Stripe pays out that transaction to the client's bank account. When the client then comes back to do the refund from Commerce, the status of the refund in Stripe get's set to pending, as they need to go and collect the funds from our customer's bank account in order to fund the refund, and then mark the refund as successful.

As far as I can tell, when it returns the refund status as Pending, Commerce marks this as a failed refund.

Really here we it to be marked as pending in Craft, perhaps offer a reason why, or at least direct them to the Stripe dashboard to see why it's pending. Then have the option to check the refund status again at a later date.

### Steps to reproduce

This is a hard one to reproduce, as from what I can see there are no Stripe test cards that simulate this scenario, I'm looking into it further though. In simple terms though, Charge the card, wait for funds to be paid out then attempt the refund, this should fail immediately. You will see something like this for the status response:

```
{
    "id": "re_xxxx",
    "object": "refund",
    "amount": 109900,
    "balance_transaction": "txn_1xxxx",
    "charge": "ch_xxxxxxx",
    "created": 1606977516,
    "currency": "aud",
    "metadata": [],
    "payment_intent": "pi_xxxxxxx",
    "reason": null,
    "receipt_number": "3057-3533",
    "source_transfer_reversal": null,
    "status": "pending",
    "transfer_reversal": null
}
```
### Additional info

- Craft version: 3.5.15.1
- PHP version: 7.3.25
- Plugins & versions: 3.2.12

#129 covers what I would like to see, Commerce checking on the status of Stripe transactions. It's all got quite out of sync now.

**Comment by lukeholder** [comment]
Confirmed this is an issue. Will support pending refunds, and  will use webhooks to update transactions when refunds are completed.

**Comment by lukeholder** [comment]
Believe this was fixed in 2.4.0
